### PR TITLE
Fix/1 sync people filter and search

### PR DESCRIPTION
**Pull request by YLeight**
Handle searching and filtration together

Fix: #1 

https://jam.dev/c/0c3e8498-5e22-43d5-b81c-a4a8f526af85

### CODE REVIEW

**Comment by YLeight** [comment]
Blocked by #4 & #6

**Comment by ecurrencyhodler** [comment]
Video looks good. Sorry one more thing. We have reverted our backend api calls from api.people-test->people.test. So we no longer will require the "api" in front of the call.

Can you push another commit? I think that'll trigger our github actions workflows for testing. Also in the future, you can run the tests locally as well using: `yarn run`.

**Comment by YLeight** [comment]
@ecurrencyhodler ✅

**Comment by YLeight** [comment]
Checking tests

**Comment by elraphty** [comment]
@YLeight fix Jest and Prettier issues

**Comment by ecurrencyhodler** [comment]
All checks pass. @elraphty

**Comment by YLeight** [comment]
I think is better to create new branch from new master as it was updated forcily

**Comment by YLeight** [comment]
@ecurrencyhodler @elraphty And here we go again

**Comment by YLeight** [comment]
@ecurrencyhodler @elraphty kindly reminder

**Comment by elraphty** [comment]
Great work @YLeight, it works well, but the user search by alias call is supposed to make a backend call so it can be a global search, and not an array search check `getPeople() ` in https://github.com/stakwork/sphinx-tribes-frontend/blob/ec15331f75313cfb3dd6a672e4fd4ba480b52ca0/src/store/main.ts, calling the function with a search text is the way to go, thanks.

**Comment by YLeight** [comment]
> Great work @YLeight, it works well, but the user search by alias call is supposed to make a backend call so it can be a global search, and not an array search check `getPeople() ` in https://github.com/stakwork/sphinx-tribes-frontend/blob/ec15331f75313cfb3dd6a672e4fd4ba480b52ca0/src/store/main.ts, calling the function with a search text is the way to go, thanks.

Are you about [this place](https://github.com/stakwork/sphinx-tribes-frontend/pull/7/files#diff-c3c87a137d85e13039838d3fde968008bea4cf7859033e4c9a6f4ce34c45a110R125), right?

**Comment by YLeight** [comment]
Or about [this](https://github.com/stakwork/sphinx-tribes-frontend/pull/7/files#diff-b6a1c1c1f6cccdc47ea30f8ecc5b60e3e1df80f164d6cb4c04fcd563048957b4R19)?

**Comment by YLeight** [comment]
@elraphty Or do you mean its would be better to rewrite old logic with usage of `src/people/utils/filterPeople.ts` util with real request, by passing all [language filters as query params](https://github.com/stakwork/sphinx-tribes/blob/master/frontend/app/src/store/main.ts#L187)?

**Comment by YLeight** [comment]
> @elraphty Or do you mean its would be better to rewrite old logic with usage of `src/people/utils/filterPeople.ts` util with real request, by passing all [language filters as query params](https://github.com/stakwork/sphinx-tribes/blob/master/frontend/app/src/store/main.ts#L187)?

Seems like in this case the util function is no more needed, and I can remove is as a legacy code. To avoid not recommended usage in the future.

@elraphty Are we on the same page?

**Comment by elraphty** [comment]
> > @elraphty Or do you mean its would be better to rewrite old logic with usage of `src/people/utils/filterPeople.ts` util with real request, by passing all [language filters as query params](https://github.com/stakwork/sphinx-tribes/blob/master/frontend/app/src/store/main.ts#L187)?
> 
> Seems like in this case the util function is no more needed, and I can remove is as a legacy code. To avoid not recommended usage in the future.
> 
> @elraphty Are we on the same page?

yeah, we don't need the util function for searching it will only filter 100 users, the people API endpoint already can search if a search query is passed,  you can leave the util function for just the skills filter, we will also move the filter search to the backend later, but it is not related to this issue.

**Comment by YLeight** [comment]
> the people API endpoint already can search if a search query is passed

I've used `getPeople` method from `main` store, as I can see, this method get [search query from `ui` store](https://github.com/stakwork/sphinx-tribes-frontend/blob/master/src/store/main.ts#L728).

> you can leave the util function for just the skills filter

I've got [observed `people` field from `main` store](https://github.com/stakwork/sphinx-tribes-frontend/pull/7/files#diff-b6a1c1c1f6cccdc47ea30f8ecc5b60e3e1df80f164d6cb4c04fcd563048957b4R20) which was updated by [`people` setter from `getPeople` method](https://github.com/stakwork/sphinx-tribes-frontend/blob/master/src/store/main.ts#L741)
And I applied util `filterPeople` util function for skills filtration of current search result

File: src/hooks/usePeopleFilters.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * Filter people list by search string and selected language
+ */
+export default function usePeopleFilters() {
+  const { main } = useStores();
+  const [codingLanguageFilter, setCodingLanguageFilter] =
+    useState<CodingLanguageFilter>(emptyCodingLanguageFilter);
+  let people = filterByCodingLanguage(
+    useFuse(main.people, ['owner_alias']).filter((p: Person) => !p.hide) || [],
+    codingLanguageFilter
+  );
```
Comment (YLeight) [inline src/hooks/usePeopleFilters.ts:22]: Something like this?
```suggestion
  main.getPeople({ langages: codingLanguageFilter });
```

Comment (elraphty) [reply src/hooks/usePeopleFilters.ts:22]: @YLeight this URL format should fix it `{host}/people?search=Yleight&languages=Node,Javascript,Golang`

Comment (YLeight) [reply src/hooks/usePeopleFilters.ts:22]: May be this reason `useFuse` was used

Comment (YLeight) [reply src/hooks/usePeopleFilters.ts:22]: As I can see, this is a problem in people-test api, as search doesn't work in any branch

**Comment by YLeight** [comment]
@elraphty Tricky content test id on loading conditional rendering

Ready for review

As I've noticed on a meeting, it seems like languages query string should be persistent (stored in a main store)
Since after page refreshing, people data are got from the local storage, but not from the request result, it could lead inconsistent state of ui with empty filters and filtered people data from local storage as it was with search result in https://github.com/stakwork/sphinx-tribes/issues/1437

Can check after `people-test.sphinx.chat/people` endpoing fix

P.S. Thanks for patience =) Was too excited)

**Comment by elraphty** [comment]
> @elraphty Tricky content test id on loading conditional rendering
> 
> Ready for review
> 
> As I've noticed on a meeting, it seems like languages query string should be persistent (stored in a main store) Since after page refreshing, people data are got from the local storage, but not from the request result, it could lead inconsistent state of ui with empty filters and filtered people data from local storage as it was with search result in [stakwork/sphinx-tribes#1437](https://github.com/stakwork/sphinx-tribes/issues/1437)
> 
> Can check after `people-test.sphinx.chat/people` endpoing fix
> 
> P.S. Thanks for patience =) Was too excited)

@YLeight The endpoint is up and running https://people-test.sphinx.chat/people?search=Evan, can you confirm from your end that it works?

**Comment by elraphty** [comment]
@YLeight 

Tested it, this and my findings from this Demo

https://www.loom.com/share/7dc68d5af5ab48c3a81382f770c18a6c

- You can remove the toLowercase on the search input, the backend handles it now
- The Skills are not making any request to the backend, the updated endpoint  https://people-test.sphinx.chat/people?search=Evan&languages=Golang,Lightning can search for both the skills and languages at the same time.

**Comment by YLeight** [comment]
Now I am able to test. I see side effects too
I'll refactor this part then.
Preparing other solution, based on request result.

**Comment by YLeight** [comment]
@elraphty https://github.com/stakwork/sphinx-tribes-frontend/issues/137 this bug partially leads problems with functionality of this PR

**Comment by ecurrencyhodler** [comment]
Hi Yegor! Looks like some tests are still failing. Can you take a look? It might help to also rebase and squash. 

Just in case you're not familiar with how that works, you can ask @Vayras or @aliraza556. Thanks.

**Comment by YLeight** [comment]
Something happen with nock
request to http://localhost:5002/people?page=1&resetPage=true&languages=&search=&sortBy=last_login&limit=500 failed, reason: Nock: No match for request

**Comment by YLeight** [comment]
Seems like backend is again unavailable

**Comment by YLeight** [comment]
> It might help to also rebase and squash.

I do this every day already)

**Comment by YLeight** [comment]
Awesome, languages now are also a part of url

Thanks to @elraphty 

As discussd with @elraphty

I am going to make interactive rebase of the commit history
Make separate PR with husky fix

During this PR were fixed:
- https://github.com/stakwork/sphinx-tribes-frontend/issues/1
- https://github.com/stakwork/sphinx-tribes-frontend/issues/137
Refactored:
- https://github.com/stakwork/sphinx-tribes/issues/1409
- https://github.com/stakwork/sphinx-tribes/issues/1437

**Comment by YLeight** [comment]
Today, this request `https://people-test.sphinx.chat/people?page=1&resetPage=true&languages=Typescript,Lightning&search=ec&sortBy=last_login&limit=500` responds with incorrect data
Attached video:
https://jam.dev/c/74573792-2400-4918-b550-2bbae443208a

**Comment by YLeight** [comment]
Implemented uploading synced with filter and search result

**Pull request closed without merging**